### ISSUE TITLE
attempt to quash a warning by storing the result of write() in _sigchld.c is unsuccessful

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5734]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#5734|
|Type|enhancement|
|Created|2012-06-30 05:25:18Z|
|Branch|https://github.com/twisted/twisted/tree/store-write-result-sigchld-5734|

We should apply this:

```
#!diff
===================================================================
--- twisted/internet/_sigchld.c	(revision 34656)
+++ twisted/internet/_sigchld.c	(working copy)
@@ -12,7 +12,7 @@
 
 static void got_signal(int sig) {
     int saved_errno = errno;
-    int ignored_result;
+    ssize_t ignored_result;
 
     /* write() errors are unhandled.  If the buffer is full, we don't
      * care.  What about other errors? */
```

since, in fact, `write()` returns `ssize_t` and not `int`.  (Sometimes they're the same thing, sometimes not.)

<details><summary>Searchable metadata</summary>

```
trac-id__5734 5734
type__enhancement enhancement
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__branches_store_write_result_sigchld_5734 branches/store-write-result-sigchld-5734
branch_author__ashfall ashfall
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1341033918000000 1341033918000000
changetime__1342350382000000 1342350382000000
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)|@ashfall set owner to @ashfall<br>@ashfall set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_3"></a>|@ashfall commented|
|-|-|

(In [34658]) Branching to 'store-write-result-sigchld-5734'

**Comment by twisted-trac** [comment]
|[<img alt="ashfall's avatar" src="https://avatars.githubusercontent.com/u/1138873?s=50" width="50" height="50">](https://github.com/ashfall)<a name="note_4"></a>|@ashfall set owner to @glyph<br>@ashfall set status to `new`|
|-|-|

Looks good to me.
Please add a news file and merge.

[Build Results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/store-write-result-sigchld-5734)

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph set status to `closed`|
|-|-|

(In [34829]) Merge store-write-result-sigchld-5734

Author: glyph

Reviewer: ashfall

Fixes: [#5734](https://github.com/twisted/twisted/issues/5734)

Fix a minor defect in `_sigchld.c`; store the result of a function in a
container declared to be of the correct size, rather than the incorrect size.
